docs-configuration.md

```diff
@@ -0,0 +1,352 @@
+# Configuration
+
+Configuration in Bigflow is managed as Python code in `bigflow.Config` objects.
+It follows the [Configuration as Code](https://rollout.io/blog/configuration-as-code-everything-need-know/) principle.
+
+You will use `bigflow.Config` to configure individual workflows as well
+as to configure GCP deployment of your project.
+
+## Config object
+
+Here is the example of a `Config`:
+
+```python
+from bigflow import Config
+
+deployment_config = Config(name='dev',
+                           properties={
+                               'gcp_project_id': '{env}-project-id',
+                               'docker_repository_project':  'my-shared-docker-project-id',
+                               'docker_repository': 'eu.gcr.io/{docker_repository_project}/my-analytics',
+                               'vault_endpoint': 'https://example.com/vault',
+                               'dags_bucket': 'europe-west1-my-1234-bucket'
+                           })\
+        .add_configuration(name='prod',
+                           properties={
+                               'dags_bucket': 'europe-west1-my-4321-bucket'
+                           })
+```
+
+`Config` is the combination of Python `dicts` with
+some extra features that are useful for configuring things:
+
+1. The `Config` object holds multiple named configurations, one configuration
+per each environment (here `dev` and `prod`).
+
+1. Properties with constant values (here `vault_endpoint`) 
```

Comment:
Properties with constant values (here `vault_endpoint` and `docker_repository_project`)